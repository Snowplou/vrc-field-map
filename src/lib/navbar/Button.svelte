<script lang="ts">
    export let text: string;
</script>

<span on:click on:keydown>
    {text}
</span>

<style>
    span {
        height: 100%;
        padding-left: 10px;
        padding-right: 10px;

        cursor: pointer;

        user-select: none;

        display: inline;

        display: flex;
        align-items: center;
        justify-content: center;

        background-color: #454545;
        color: #E0E0E0;
        border-radius: 12px;

        margin-right: 5px;
    }

    span:hover {
        background-color: #656565;
        color: #E0E0E0;
    }
</style>