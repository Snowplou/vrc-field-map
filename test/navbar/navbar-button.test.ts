import { cleanup, fireEvent, render, screen } from '@testing-library/svelte';
import { describe, it, expect, afterEach, vi } from "vitest";
import Button from '../../src/lib/navbar/Button.svelte';

describe('Button.svelte', () => {
    // TODO: @testing-library/svelte claims to add this automatically but it doesn't work without explicit afterEach
    afterEach(() => cleanup());

    it('mounts', () => {
        const { container } = render(Button, { text: "TestingTesting123" });
        expect(container).toBeTruthy();
        expect(container.innerHTML).toContain("TestingTesting123");
        expect(container.innerHTML).toMatchSnapshot();
    });

    it('registers click', async () => {
        const { component } = render(Button, { text: "TestingTesting123" });

        const fn = vi.fn();
        component.$on("click", fn);

        await fireEvent.click(screen.getByText("TestingTesting123" ));
        expect(fn.mock.calls.length).toEqual(1);
    });
});