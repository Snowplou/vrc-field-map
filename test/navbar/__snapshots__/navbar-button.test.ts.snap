// Vitest Snapshot v1

exports[`Button.svelte > mounts 1`] = `"<div><span class=\\"svelte-pbnhkd\\">TestingTesting123</span></div>"`;
